website/docs/r/nsxt_edgegateway.html.markdown

```diff
@@ -223,12 +272,29 @@ definition at provider level.
 * `description` - (Optional) A unique name for the edge gateway.
 * `external_network_id` - (Required) An external network ID. **Note:** Data source [vcd_external_network_v2](/providers/vmware/vcd/latest/docs/data-sources/external_network_v2)
 can be used to lookup ID by name.
-* `subnet` - (Required) One or more [subnets](#edgegateway-subnet) defined for edge gateway.
 * `edge_cluster_id` - (Optional) Specific Edge Cluster ID if required
 * `dedicate_external_network` - (Optional) Dedicating the External Network will enable Route Advertisement for this Edge Gateway. Default `false`.
 
+* `subnet` - (Optional) One or more [subnets](#edgegateway-subnet) defined for edge gateway. One of
+  `subnet`, `auto_subnet` or `auto_allocated_subnet` is **required**.
+* `auto_subnet` - (Optional, *v3.9+*) One or more [subnets](#edgegateway-auto-subnet) defined for
+  Edge Gateway. One of `subnet`, `auto_subnet` or `auto_allocated_subnet` is **required**.
+* `auto_allocated_subnet` - (*v3.9+*) One or more [subnets](#edgegateway-auto-allocated-subnet)
```

Comment:
No. These are completely made-up constructs. All 3 of them work with the same API fields, but their behaviors are different. We have 3 different `TypeSet` blocks because of Terraform schema limitations.

Reply:
So, we're converging towards:

`subnet`, `subnet_total_ip_count` and `subnet_ip_auto_allocation`
(complete picture with all three types)

Reply:
> Now towards... :)
> 
> `subnet`, `subnet_with_total_ip_count` and `subnet_with_ip_count`

Code (and PR description) has been changed to use this notation

Reply:
Looks great, thank you!